src/Process/PlaceOnGroundProcessing.js

```diff
@@ -0,0 +1,131 @@
+import * as THREE from 'three';
+import Coordinates from '../Core/Geographic/Coordinates';
+import Extent from '../Core/Geographic/Extent';
+import RendererConstant from '../Renderer/RendererConstant';
+import { l_ELEVATION } from '../Renderer/LayeredMaterial';
+
+function tileAt(pt, tile) {
+    if (tile.extent) {
+        if (!tile.extent.pointIsInside(pt)) {
+            return undefined;
+        }
+
+        for (const c of tile.children) {
+            const t = tileAt(pt, c);
+            if (t) {
+                return t;
+            }
+        }
+        if (tile.getLayerTextures(l_ELEVATION)[0].coords.zoom > -1) {
+            return tile;
+        }
+        return undefined;
+    }
+}
+
+let _canvas;
+function _readTextureValueAt(layer, texture, x, y) {
+    if (texture.image.data) {
+        return texture.image.data[y * texture.image.width + x];
+    } else {
+        if (!_canvas) {
+            _canvas = document.createElement('canvas');
+            _canvas.width = 1;
+            _canvas.height = 1;
+        }
+        const ctx = _canvas.getContext('2d');
+        ctx.drawImage(texture.image, x, y, 1, 1, 0, 0, 1, 1);
+        const d = ctx.getImageData(0, 0, 1, 1);
+        return THREE.Math.lerp(
+            layer.materialOptions.colorTextureElevationMinZ,
+            layer.materialOptions.colorTextureElevationMaxZ,
+            d.data[0] / 255);
+    }
+}
+
+function _updateSinglePoint(referenceCrs, layer, nodes, crs, toModify, zOffset) {
+    const pt = new Coordinates(referenceCrs, toModify).as(crs);
+    pt._values[2] = 0;
+
+    let tileWithValidElevationTexture = null;
+    for (const n of nodes) {
+        tileWithValidElevationTexture = tileAt(pt, n);
+        if (tileWithValidElevationTexture) {
+            break;
+        }
+    }
+
+    if (tileWithValidElevationTexture) {
+        const src = tileWithValidElevationTexture.getLayerTextures(l_ELEVATION)[0];
+        const width = src.image.width;
+        const height = src.image.height;
+
+        const tmp = new Extent(pt.crs, pt, pt);
+
+        // Assuming that tiles are split in 4 children, we lookup the parent that
+        // really owns this texture
+        const stepsUpInHierarchy = Math.log2(1.0 /
+            tileWithValidElevationTexture.materials[RendererConstant.FINAL].offsetScale[l_ELEVATION][0].z);
+        for (let i = 0; i < stepsUpInHierarchy; i++) {
+            tileWithValidElevationTexture = tileWithValidElevationTexture.parent;
+        }
+
+        const offset = tmp.offsetToParent(tileWithValidElevationTexture.extent);
```

Comment:
No we need the one that's displayed (I need to fix this as well).
We want to get the same result that the shader - so we need to sample at the same point (and the current logic is broken, since we should interpolate between the 3 vertices of the trianges used to rasterize the given point)...